homeassistant/components/sunweg/coordinator.py

```diff
@@ -0,0 +1,105 @@
+"""The Sun WEG plant coordinator integration."""
+
+import datetime
+import logging
+
+from sunweg.api import APIHelper, LoginError, SunWegApiError
+from sunweg.plant import Plant
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_PLANT_ID, DOMAIN, DeviceType
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SunWEGDataUpdateCoordinator(DataUpdateCoordinator[Plant]):
+    """SunWEG Data Update Coordinator coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, api: APIHelper, plant_id: int, plant_name: str
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="SunWEG sensor",
+            update_interval=SCAN_INTERVAL,
+        )
+        self.api = api
+        self.plant_id = plant_id
+        self.plant_name = plant_name
+        self.device_info = DeviceInfo(
```

Comment:
This is usually just set in the entity itself, not sure it needs to be in the coordinator

Reply:
I saw that in an example, I don't know how to move to the sensor itself.

Reply:
You should just be able to set `self._attr_device_info = DeviceInfo(` in the sensor.py.